### PR TITLE
Enable warnings as errors in compiler component on S390

### PR DESCRIPTION
**Pull request by fjeremic**
Address the last warning on z/OS and enable warnings as errors on all S390 platforms.

### CODE REVIEW

**Comment by fjeremic** [comment]
Leaving this in WIP state until I've ran all the testing possible to ensure we don't get build breaks from warnings being treated as errors.

**Comment by fjeremic** [comment]
Jenkins compile all jdk8 depends eclipse/omr#6108

**Comment by fjeremic** [comment]
Jenkins test sanity all jdk8 depends eclipse/omr#6108

**Comment by fjeremic** [comment]
Jenkins test sanity win32 jdk8 depends eclipse/omr#6108

**Comment by fjeremic** [comment]
Jenkins test sanity plinux jdk8 depends eclipse/omr#6108

**Comment by fjeremic** [comment]
Jenkins test sanity plinux jdk8 depends eclipse/omr#6108

**Comment by fjeremic** [comment]
PPCLE Linux failures are due to https://github.com/eclipse-openj9/openj9/issues/13313#issuecomment-899728712. We will need to wait until the fix is merged before these start passing.

**Comment by fjeremic** [comment]
Rebased to pick up the above fix.

Jenkins test sanity all jdk8 depends eclipse/omr#6108

**Comment by fjeremic** [comment]
Tested all the changes internally and externally. All tests are passing. Removing WIP as this is ready for merge. Note the dependency on OMR requiring a coordinated merge.

**Review by dsouzai — APPROVED** [review]

File: debugtools/DDR_VM/data/superset-constants.dat
Comment (keithc-ca) [inline debugtools/DDR_VM/data/superset-constants.dat:4600]: Additions should **not** be made to this file (`MM_LightweightNonReentrantRWLock` was not previously present).

Comment (keithc-ca) [reply debugtools/DDR_VM/data/superset-constants.dat:4600]: Fixed via #13417.

**Pull request merged**
